### PR TITLE
Add Arduino CLI v0.16.0 with support for basic commands

### PR DESCRIPTION
**Pull request by zmoog**
### Motivation

I want to build a simple and straightforward example of an Alfred Workflow that can wrap a 3rd party tool and make it available through Alfred for convenience.

### Change description

Added a simple handler with some flexibility build in to run the Arduino CLI binary (embedded in the project), parse the JSON response and build the response to render the data on the Alfred UI.

- [x] Notarize the official `arduino-cli` binary with my personal Apple Developer ID.

### Additional Notes

This project will be used as an example on my personal blog at https://zmoog.dev/post/alfred-workflow.

### Reviewer checklist

* [ ] PR address a single concern.
* [ ] PR title and description are properly filled.
* [ ] Changes will be merged in `master`.
* [ ] Changes are covered by tests.
* [ ] Logging is meaningful in case of troubleshooting.
* [ ] History is clean, commit messages are meaningful (see `CONTRIBUTING.md`) and are well formatted.

### CODE REVIEW

File: arduino-cli.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    result = run(cmd, hide=True, warn=True)
+    if not result.ok:
+        raise Exception(result.stdout)
+
+    return json.loads(result.stdout)
+
+
+def search_key_for_board(board):
+    elements = []
```
Comment (xrmx) [inline arduino-cli.py]: You can define the list with the content without the following append

File: arduino-cli.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    elements.append(board["name"])
+    elements.append(board["FQBN"])
+    return u" ".join(elements)
```
Comment (xrmx) [inline arduino-cli.py]: No need for u

File: arduino-cli.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+
+def run_command(cmd):
+    from invoke import run
+    
+    cmd = "./arduino-cli {cmd} --format json".format(cmd=cmd)
+
+    result = run(cmd, hide=True, warn=True)
+    if not result.ok:
+        raise Exception(result.stdout)
```
Comment (xrmx) [inline arduino-cli.py]: Better use a specialized exception

File: arduino-cli.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    def handle_core_list(self):
+
+        def wrapper():
+            return run_command("core list")
+
+        cores = self.wf.cached_data('cores', wrapper, max_age=120)
+
+        if self.args.query:
+            cores = self.wf.filter(self.args.query,
+                                   boards,
```
Comment (xrmx) [inline arduino-cli.py:84]: cores?

File: arduino-cli.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    # add an optional query and save it to 'query'
+    parser.add_argument('query', nargs='?', default=None)
+
+    # parse the script's arguments
+    args = parser.parse_args(wf.args)
+
+    Handler(args, wf).run()
+
+
+if __name__ == u"__main__":
```
Comment (xrmx) [inline arduino-cli.py:195]: u

**Review by xrmx — COMMENTED** [review]
.

File: arduino-cli.py (hunk 1 above)
Comment (zmoog) [reply arduino-cli.py]: Appends are not welcome here!

File: arduino-cli.py (hunk 2 above)
Comment (zmoog) [reply arduino-cli.py]: Honestly, I don't remember why I added them, but are non required indeed.. thanks!

File: arduino-cli.py (hunk 3 above)
Comment (zmoog) [reply arduino-cli.py]: I added two specialized exceptions. 🙇

File: arduino-cli.py (hunk 4 above)
Comment (zmoog) [reply arduino-cli.py:84]: Fixed! 🙇

File: arduino-cli.py (hunk 5 above)
Comment (zmoog) [reply arduino-cli.py:195]: Gone.

**Pull request merged**